lib/SIL/Verifier/SILVerifier.cpp

```diff
@@ -5271,6 +5271,13 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
             "convention");
   }
 
+  void checkMoveValueInst(MoveValueInst *mvi) {
+    require(mvi->getOperand()->getType().isObject(),
+            "Operand value should be an object");
+    require(mvi->getType() == mvi->getOperand()->getType(),
+            "Result and operand must have the same type, today.");
```

Comment:
Clearly this type restriction needs to be relaxed once you have moveOnly SIL types